Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -147,6 +151,16 @@
 
 % We skip the computation of PWI4D if it is provided
 if nargin>=3 && ~isempty(PWI4D)
+	if ischar(PWI4D) && xASL_exist(PWI4D, 'file')
```

Comment:
Actually - not really. xASL_io_ASLSubtractionAveraging loads and saves everything...
We only want to load and average stuff, not to save everything. 

So thats why xASL_im_ASL was better. Actually, xASL_im_ASL is used already independently on many places with ASL4D as input. And here I only needed to average things so that I could run it for despiked_ASL. I only wanted a correct average of despiked.nii and not create and save despiked_PWI3D etc.

Reply:
The way I designed it, you have the following choices (on purpose):

1. We always use xASL_io_ASLSubtractionAveraging, because you always load and save NIfTIs/JSONs. We don't want to put stuff back to memory anymore.
2. For (despiked_)ASL4D only, we currently have an exception, because we want to be able to create a temporary PWI from the raw ASL4D with motion application, as temporary PWI.

What you changed, is that you can also input non-ASL4D paths to xASL_im_ASLSubtractionAveraging, which is a behavior we don't want. So please revert this, and use one of the 2 options above.

Do you agree with this logic?

Reply:
Note that I have changed this behavior on all occurrences, so I closed all the other conversations because I change it everywhere how you asked.